### DISCUSSION TITLE
Option to log messages

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#915] [Q&A]
Hi, 
is there an option to log the messages the server receive and sends into a log file
Thanks

**Comment by msujew** [comment]
Hey @cdietrich,

there isn't one in Langium itself, because vscode already provides that (assuming the underlying vscode extension uses the `vscode-languageclient`). See [this vscode documentation](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#logging-support-for-language-server).

**Reply by cdietrich** [reply]
hmm i have the problem that i dont see published diagnostics if i run the language server in intellij. this is why i am asking.

**Reply by cdietrich** [reply]
will try to patch that idea lsp plugin to log there

**Reply by cdietrich** [reply]
i can see other server responses but i cannot see the publishDiagnostics one.

**Reply by msujew** [reply]
Are you only listening to requests or also on notifications? `publishDiagnostics` is only a notification to the language client.

**Reply by cdietrich** [reply]
I am using this one
https://github.com/ballerina-platform/lsp4intellij
am not sure if notifications are broken there
Will do a counter Test with Xtext based server

**Reply by cdietrich** [reply]
its also not working with xtext.

**Reply by cdietrich** [reply]
created https://github.com/ballerina-platform/lsp4intellij/issues/303
